api/src/utils/apply-query.ts

```diff
@@ -540,22 +540,23 @@ export function applyFilter(
 			// See https://github.com/knex/knex/issues/4518 @TODO remove as any once knex is updated
 
 			// These operators don't rely on a value, and can thus be used without one (eg `?filter[field][_null]`)
-			if (operator === '_null' || (operator === '_nnull' && compareValue === false)) {
-				dbQuery[logical].whereNull(selectionRaw);
+			const defaultValue = compareValue === false ? false : true;
```

Comment:
Well if it is not failing validation, why the explicit use of Boolean(compareValue) vs just compareValue?

Reply:
This actually breaks the rest of the code, since you are overwriting compareValue which is being used later on in the case where operator is not "_null | _empty …" So need to keep it being compareValue instead of casting too Boolean.